test/performance/cicd/perf-test.go

```diff
@@ -0,0 +1,91 @@
+/*
+Copyright 2024 The KubeStellar Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package perf
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/onsi/ginkgo/v2"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	ksapi "github.com/kubestellar/kubestellar/api/control/v1alpha1"
+	"github.com/kubestellar/kubestellar/test/util"
+)
+
+const (
+	ns = "nginx"
+)
+
+var _ = ginkgo.Describe("performance tests", func() {
+	ginkgo.BeforeEach(func() {
+		util.CleanupWDS(ctx, wds, ksWds, ns)
+	})
+
+	ginkgo.Context("scalability", func() {
+		ginkgo.It("create a bindingpolicy and a single deployment", func() {
+			const numDeployments = 10
+			sumBinding, sumManifestwork, sumDeploymentWDS, sumDeploymentWEC, sumTotal := 0, 0, 0, 0, 0
+
+			for i := 0; i < numDeployments; i++ {
+				util.CreateBindingPolicy(ctx, ksWds, "performance",
+					[]metav1.LabelSelector{
+						{MatchLabels: map[string]string{"name": "cluster1"}},
+					},
+					[]ksapi.DownsyncObjectTest{
+						{ObjectSelectors: []metav1.LabelSelector{
+							{MatchLabels: map[string]string{"app.kubernetes.io/name": "nginx"}},
+						}}})
+				start := time.Now()
+				util.CreateDeployment(ctx, wds, ns, fmt.Sprint(i),
+					map[string]string{
+						"app.kubernetes.io/name": "nginx",
+					})
+				util.ValidateNumDeployments(ctx, wec1, ns, 1)
+
+				timeDeploymentWDS := int(util.GetDeploymentTime(ctx, wds, ns, fmt.Sprint(i)).Sub(start).Seconds())
```

Comment:
`util.GetDeploymentTime` can take a measurable amount of time, since it makes a call on the apiserver. I would do such work after the work being measured.